tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,253 @@ int main(int argc, char **argv)
+                .allow_chacha20_boosting = true
+            };
+
+            const struct s2n_security_policy test_chacha20_and_boosted = {
+                .minimum_protocol_version = S2N_SSLv3,
+                .cipher_preferences = &cipher_preference_with_chacha20_and_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            const struct s2n_security_policy test_tls12_chacha20_and_boosted = {
+                .minimum_protocol_version = S2N_SSLv2,
+                .cipher_preferences = &cipher_preference_with_tls12_chacha20_and_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            const struct s2n_security_policy test_chacha20_and_not_boosted = {
+                .minimum_protocol_version = S2N_SSLv3,
+                .cipher_preferences = &cipher_preference_with_chacha20_and_no_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            const struct s2n_security_policy test_no_chacha20_and_boosted = {
+                .minimum_protocol_version = S2N_SSLv3,
+                .cipher_preferences = &cipher_preference_without_chacha20_and_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            uint8_t chacha20_boosted_wire[] = {
+                TLS_CHACHA20_POLY1305_SHA256,
+                TLS_AES_256_GCM_SHA384,
+                TLS_AES_128_GCM_SHA256,
+            };
+
+            EXPECT_SUCCESS(s2n_enable_tls13_in_test());
+            /* If the server chose a security policy w/ ChaCha20 AND has ChaCha20 boosting enabled
+             * then negotiate ChaCha20 iff the client has ChaCha20 as the most-preferred cipher.
+             */
+            {
+                conn->security_policy_override = &test_chacha20_and_boosted;
+                conn->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+                conn->client_protocol_version = S2N_TLS13;
+                conn->actual_protocol_version = S2N_TLS13;
+                conn->server_protocol_version = S2N_TLS13;
+
+                uint8_t count = sizeof(chacha20_boosted_wire) / S2N_TLS_CIPHER_SUITE_LEN;
+                EXPECT_SUCCESS(s2n_set_cipher_as_tls_server(conn, chacha20_boosted_wire, count));
+
+                if (s2n_chacha20_poly1305.is_available()) {
```

Comment:
Why not gate most of these tests on `s2n_chacha20_poly1305.is_available()`, and only have one that tests the case where chacah20 isn't available?